pwncat/commands/__init__.py

```diff
@@ -97,6 +98,31 @@ class Complete(Enum):
     """ Do not provide argument completions """
 
 
+class ParseType(Enum):
+    """
+    Command type. This defines how command parameter arguments are parsed
+    """
+
+    NONE = auto()
+    """ No specific type given, so no interpreter needed """
+    LOCAL_FILE = auto()
+    """ File type """
+
+
+class ParameterParse:
+    def parse(value):
+        """This is where the parameter will be parsed"""
+        raise NotImplementedError
+
+
+class LocalFileParse:
+    def parse(value):
+        if value.startswith("~"):
```

Comment:
Shouldn't we use the following?
```python
    if value.startswith("~/"):
```

So that if an attacker tries to upload `~tempfile` or similar, we parse the argument correctly.

Reply:
On a second thought, you may want to use
```python
    value = os.path.expanduser(value)
```

> NOTE: this will replace `~user`, eg. `~root` with `/root`

Reply:
```
    if value.startswith("~/"):

Reply:
```
    if value.startswith("~/"):
```
Isn't always possible since the trailing `/` gets sliced off in the completer. The `expanduser` is a good idea.